README.md

```diff
@@ -1 +1,53 @@
-# VTEX Checkout
+# VTEX Checkout 1.x
+
+Vtex IO port to SmartCheckout. 
+
+## Building the checkout page
+
+The class responsible for building the page react will render is `HtmlPageBuilder`. 
+
+### Head
+
+The head tags received from `checkoutHtml`'s graphQL query are converted to `React` elements for insertion via `Helmet`.
+Before doing this some elements from the original head are filtered out. At the moment only the script for `render-extension-loader`
+is removed. This head building is done in the function `getHelmetHead`.
+
+In addition to these elements, a script defined at `react/core/scripts.tsx` is inserted in the head as well: `addTachyonsScoped`. 
+Its behavior is to add a `link` tag with a `href` to a scoped tachyons stylesheet.
+
+It's important to note that apparently `script` tags added via `Helmet` receive an additional `defer` prop.
+
+### Body 
+
+The `getBody` function is responsible for building the renderable part of the app:
+```
+<div className="checkout-app">
+  <script dangerouslySetInnerHTML={ { __html: deleteTachyons } }/>
+  <div dangerouslySetInnerHTML={ { __html: this.body } } />
+  { this.getBodyScripts() }
+</div>
+```
+
+The body from checkout's html is inserted in via `dangerouslySetInnerHTML`. Right before it a script is inserted for deleting
+the unscoped tachyons inserted by the `render-server`. After them all body scripts present in the original checkout html are
+inserted. 
+
+Some notes about these body scripts is that a `defer` prop is added to all of them. Besides this, in the function `getBodyScripts`
+some scripts are replaced. At the moment the scripts replaced are:
+
+- **render-loader-script**: This inline script originally depends on `render-extension-loader`. It defines an interface for `checkout.min.js` to use.
```

Comment:
I think would be helpful to explain why you're doing this before explaining how. When explaining how, it would be nice to go about drilling down complexity, that is, explain it generally, then go into specifics.